core/src/main/scala/kafka/server/Server.scala

```diff
@@ -69,13 +69,7 @@ object Server {
   ): KafkaMetricsContext = {
     val contextLabels = new java.util.HashMap[String, Object]
     contextLabels.put(ClusterIdLabel, clusterId)
-
-    if (config.usesSelfManagedQuorum) {
-      contextLabels.put(NodeIdLabel, config.nodeId.toString)
-    } else {
-      contextLabels.put(BrokerIdLabel, config.brokerId.toString)
-    }
-
+    contextLabels.put(BrokerIdLabel, config.brokerId.toString)
```

Comment:
Hmm, this seems like a bug - we may want to add a unit test if nothing failed. We should be setting `NodeIdLabel`, not `BrokerIdLabel`. Right?

Reply:
In [ServerTest.scala](https://github.com/apache/kafka/pull/18547/files#diff-65b84c35e763043ec7de1cb79d789ce336262cd27cdc63502cdf2b7cb4553296) will test this bug, I think we won't need to add a new test for it.